web-console/knikubevirt/snapshots/snapshots.md

```diff
@@ -0,0 +1,72 @@
+# Virtual Machines Snapshots
+
+Snapshots allow users to create a copy of the VM disks and memory, mainly for restore and backup purposes. The snapshots are taken live or offline depending on if the VM is running or not. 
+This document describes the snapshot design flow.  
+
+
+
+
+## From VM List 
+
+![Offline VM list - snapshot action from a single VM kebab](img/1-00.jpg)
+When clicking on the kebab menu in the VM list, the user will be able to take a snapshot of the VM, meaning he will copy the disks, memory content, and metadata for restoring, backup, creating templates and cloning VMs.
+
+![Offline Snapshot modal - default](img/1-01.jpg)
+In the basic modal, the snapshot will take all current configuration and selected parameters.
+The user must add a descriptive name.
+![Offline Snapshot modal - selecting disks](img/1-02.jpg)
+In the disks field, the user will be able to decide if to include all disks or just some of them and if to include memory as part of the snapshot or not.
+![Offline Snapshot modal - disks customization](img/1-03.jpg)
+![Offline Snapshot modal - disks customized](img/1-04.jpg)
+
+![Offline Snapshotting status indication](img/1-05.jpg)
+![Offline Snapshotting status indication on click and kebab mode](img/1-06.jpg)
+
+Any virtual machine that has a snapshot in progress will show this in the state:
+When in this progress mode all actions would be disabled and the “Stop snapshot” would be available.
+
+
+## VM page - snapshots tab
+
+![Offline Shanpshot tab](img/2-00.jpg)
```

Comment:
I can't think of any cases where we have multiple levels of tabs today. Doesn't mean it isn't possible, but wondering if this is more of a filter instead of tabs. Are we just filtering between past and present (scheduled) snapshots? What about using the toggles instead?

Reply:
@beanh66 
I think that these are two different lists altogether. Each one displays different entities and has different columns.
I'm ok with using any other UI element that allows both these lists to coexist.
I think using a toggle button is a bit confusing next to the main action (see attached image).
Using sections is ok but less effective on long lists (see attached image).

WDYT? should I change the tabs to any of these?
![2-00 B](https://user-images.githubusercontent.com/48942365/59269989-0d063780-8c59-11e9-8aa0-fa35c6f959ea.jpg)
![2-00 C](https://user-images.githubusercontent.com/48942365/59269990-0d063780-8c59-11e9-8245-1a5e4a07afe2.jpg)

Reply:
If we think of Scheduled Snapshots as basically Snapshots with a future "Date", there might be a way to combine the lists.

I'm not saying this is perfect, but it would at least avoid this odd tab/toggle/split design for now. I'm not a fan of the toggle - the two separate lists on the same page is okay, but as Ran said there are scaling issues.

Snapshot items include columns for:
- Name
- Date
- CPU (which I suggest we remove)
- Size (which could be greyed out "None" for scheduled snapshots)
- Memory (also greyed out "None" for scheduled snapshots)
- Disks (stay the same for scheduled snapshots)
- Networks (which I also suggest we remove)

Scheduled Snapshots include columns for:
- Type
- Day (related to Date)
- Time Zone (related to Date)
- Backup Policy

Right now Scheduled Snapshots have no column describing what's being snapshotted, making them impossible to distinguish between. That's an issue, and the solution is probably adding a Name column. The schedule snapshot's name would probably be the VM name with some numbers at the end.

That would make scheduled snapshots appear in the list directly next to the snapshots they created (since this is alphabetized by default). The Date column would then include the future date of the next scheduled snapshot. The Type and Backup Policy columns would probably need to be removed, though.

Reply:
I like the idea of combining the lists as well if possible, but I will leave that up to your @Ranelim and @andybraren since you know more about this resource type. If we had to keep separate I tend to like the second option with the horizontal separator. Unsure on the scale question as well. How many snapshots will be in here? If this list will typically get very lengthy then maybe the original tabs are inevitable. @lizsurette @matthewcarleton thoughts?

Reply:
@beanh66 ya scale is a big factor here. @Ranelim can you consult with dev on this first? That should really help guide the design here.

Reply:
Sure, no prob. I'll create a combined version too just so we can compare Sections on the same level

Reply:
@Ranelim maybe think of going back to the previous design where we had an indication saying if it's scheduled or not. 
![snapshots 1](https://user-images.githubusercontent.com/44394257/59560878-64bded80-9021-11e9-8d1a-2ac77f104594.png)

Reply:
The issue that we're trying to solve is avoiding secondary navigation between two different lists.
I think they shouldn't be combined since the first thing the user needs to understand is what has happened and what will happen in the future. I believe this image is the common ground between splitting the lists and combining them. 
![2-00](https://user-images.githubusercontent.com/48942365/59561426-c6ce2100-9028-11e9-9d47-7ccb22e1a3dd.jpg)

I started by creating a collapse/expand mode for each list (next to the list title) but it didn't solve the problem since the user either will not see the "Snapshots taken" in case of a long "scheduled" list, OR not see "Scheduled" if it is collapsed. So I went with "Show all.../Show less" button

Reply:
@Ranelim Did you explore the combined table view that Andy suggested? Curious to know how that looked.

Reply:
I did. Looking at it again, I admit that this is better.
![2-02a](https://user-images.githubusercontent.com/48942365/59741303-4a7b4e00-9273-11e9-8372-808ca850cf39.jpg)
I'll update this PR with this

Reply:
Thanks for making this update Ran, the latest commit looks good to me and seems like a workable compromise.